modular_citadel/code/game/machinery/toylathe.dm

```diff
@@ -0,0 +1,389 @@
+#define AUTOYLATHE_MAIN_MENU       1
+#define AUTOYLATHE_CATEGORY_MENU   2
+#define AUTOYLATHE_SEARCH_MENU     3
+
+/obj/machinery/autoylathe
+	name = "autoylathe"
+	desc = "It produces items using plastic, metal and glass."
+	icon_state = "autolathe"
+	density = TRUE
+	anchored = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 10
+	active_power_usage = 100
+	circuit = /obj/item/circuitboard/machine/autoylathe
+
+	var/operating = FALSE
+	var/list/L = list()
```

Comment:
don't initialize non var/static lists in the definition. put it in Initialize.
this is because putting in definitions will make byond call a hidden init proc which is 2x proccall overhead as doing it in Initialize().

Reply:
This is still all kinda new here. I quite literally copied the autolathe code. Most of it is the same as it is here.

Reply:
How should I fix that?

Reply:
Is this what you meant @kevinz000 ?

Reply:
Nope, still clueless. I tried moving all the vars below the initialize but that just broke things.

Reply:
sorry i fell asleep
anyways take out the = list()s from the section (here) where you define variables for the thing
move it to Initialize()
so Initialize() will have L = list() to initialize the lists.
hell, make them lazy lists (I can give a tutorial on how list inits and lazy lists work on discord, I just got up.)